src/gazebo_ros_interface.cpp

```diff
@@ -0,0 +1,377 @@
+          std::vector<ros::Publisher> imu_ros_publishers = {ros_imu_pub};
+
+          // handle IMU plugin data
+          ConnectHelper<sensor_msgs::msgs::Imu>(&GazeboRosInterface::GzImuMsgCallback, this, "",
+            "~/" + models[i]->GetName() + "/imu", imu_ros_publishers, gz_node_handle_);
+        }
+        // Verify if the Baro plugin is loaded
+        if (plugins.plugins(j).filename() == "libgazebo_barometer_plugin.so") {
+          // Create ROS publishers (needs to be an array since the same callback handles multiple publishers)
+          ros::Publisher pressure_pub = ros_node_handle_->advertise<sensor_msgs::FluidPressure>("/gazebo/" + models[i]->GetName() + "/baro/abs_pressure", 1);
+          ros::Publisher pressure_alt_pub = ros_node_handle_->advertise<geometry_msgs::Vector3Stamped>("/gazebo/" + models[i]->GetName() + "/baro/pressure_alt", 1);
+          ros::Publisher temperature_pub = ros_node_handle_->advertise<sensor_msgs::Temperature>("/gazebo/" + models[i]->GetName() + "/baro/temperature", 1);
+          std::vector<ros::Publisher> baro_ros_publishers = {pressure_pub, pressure_alt_pub, temperature_pub};
+
+          // handle Baro plugin data
+          ConnectHelper<sensor_msgs::msgs::Pressure>(&GazeboRosInterface::GzBaroMsgCallback, this, "",
+            "~/" + models[i]->GetName() + "/baro", baro_ros_publishers, gz_node_handle_);
+        }
+      }
+    }
+
+    // Get model joints in order to check for sensors
+    // Probably not the best way but the one that does work for now
+    auto joints = models[i]->GetJoints();
+    std::vector<std::string> joint_names;
+    for(size_t y = 0; y < joints.size(); y ++) {
+      joint_names.push_back(joints[y]->GetName());
+    }
+
+    // Verify if the IRLock sensor joint exists
+    if (std::find(joint_names.begin(), joint_names.end(), "irlock_joint") != joint_names.end()) {
+      // Create ROS publisher
+      ros::Publisher irlock_pub = ros_node_handle_->advertise<mavlink_sitl_gazebo::IRLock>("/gazebo/" + models[i]->GetName() + "/irlock/data", 1);
+      std::vector<ros::Publisher> irlock_ros_publishers = {irlock_pub};
+
+      // handle Diff pressure plugin data
+      ConnectHelper<sensor_msgs::msgs::IRLock>(&GazeboRosInterface::GzIRLockMsgCallback, this, "",
+        "~/" + models[i]->GetName() + "/camera/link/irlock", irlock_ros_publishers, gz_node_handle_);
+    }
+    // Verify if the Lidar (or model specific) sensor joint exists
+    if (std::find(joint_names.begin(), joint_names.end(), "lidar_joint") != joint_names.end() ||
+        std::find(joint_names.begin(), joint_names.end(), "sf10a_joint") != joint_names.end()) {
+      // Create ROS publisher
+      ros::Publisher lidar_pub = ros_node_handle_->advertise<sensor_msgs::Range>("/gazebo/" + models[i]->GetName() + "/lidar/data", 1);
+      std::vector<ros::Publisher> lidar_ros_publishers = {lidar_pub};
+
+      // handle Diff pressure plugin data
```

Comment:
What is the relation between diff pressure and the laser scanner? I don't understand that.
Or is that comment just outdated?

Reply:
ah sorry. outdated